src/components/Tree/TreeItem.tsx

```diff
@@ -0,0 +1,139 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, { Fragment, ReactElement, ReactNode, useState } from 'react';
+import { useDrag } from 'react-dnd';
+
+import { DropZone } from '@components/DropZone';
+import { TreeItemProps } from '@components/Tree/types';
+import { useTreeContext } from '@components/Tree/TreeContext';
+import { DraggableItem, DropZonePosition } from '@utilities/dnd';
+import { merge } from '@utilities/merge';
+
+export const TreeItem = ({ id, sort, onNodeSelect, onDrop, component, children }: TreeItemProps) => {
+    const { treeId, selectedIds, onSelect, draggable } = useTreeContext();
+
+    const [showNodes, setShowNodes] = useState<boolean>(false);
+
+    const [{ opacity }, drag] = useDrag({
+        item: { id, sort },
+        collect: (monitor) => ({
+            opacity: monitor.isDragging() ? 0.4 : 1,
+        }),
+        type: treeId,
+        canDrag: draggable,
+    });
+
+    const handleDrop = (
+        targetItem: DraggableItem<{ id: string; sort: Nullable<number> }>,
+        sourceItem: DraggableItem<{ id: string; sort: Nullable<number> }>,
+        position: DropZonePosition,
+    ) => {
+        onDrop?.(targetItem, sourceItem, position);
+    };
+
+    const handleSelect = () => {
+        onSelect(id);
+        onNodeSelect?.(id);
+    };
+
+    const toggleNodesVisibility = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+        event.stopPropagation();
+        setShowNodes(!showNodes);
+    };
+
+    const childrenArray = React.Children.toArray(children);
+
+    let enhancedChildren: ReactNode = childrenArray;
+
+    if (draggable) {
+        enhancedChildren = React.Children.map(children, (child, index) => {
+            if (!child) {
+                return <></>;
+            }
+
+            return React.cloneElement(
+                <Fragment key={index}>
+                    {index === 0 && (
+                        <DropZone
+                            data={{
+                                targetItem: { id, sort },
+                                position: DropZonePosition.Before,
+                            }}
+                            onDrop={handleDrop}
+                            treeId={treeId}
+                        />
+                    )}
+
+                    {child as ReactElement}
```

Comment:
We should only allow a child of type TreeItem to be passed to TreeItem or Tree component.